admin-events/src/test-utils.js

```diff
@@ -0,0 +1,220 @@
+const BigNumber = require('bignumber.js');
+const { ethers } = require('forta-agent');
+const utils = require('./utils');
+
+const defaultTypeMap = {
+  uint256: 0,
+  'uint256[]': [0],
+  address: ethers.constants.AddressZero,
+  'address[]': [ethers.constants.AddressZero],
+  bytes: '0xff',
+  'bytes[]': ['0xff'],
+  bytes32: 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+  'bytes32[]': [0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF],
+  string: 'test',
+  'string[]': ['test'],
+};
+
+function getObjectsFromAbi(abi, objectType) {
+  const contractObjects = {};
+  abi.forEach((entry) => {
+    if (entry.type === objectType) {
+      contractObjects[entry.name] = entry;
+    }
+  });
+  return contractObjects;
+}
+
+function getEventFromConfig(abi, events) {
+  let eventInConfig;
+  let eventNotInConfig;
+  let findingType;
+  let findingSeverity;
+
+  const eventsInConfig = Object.keys(events);
+  const eventObjects = getObjectsFromAbi(abi, 'event');
+  Object.keys(eventObjects).forEach((name) => {
+    if ((eventNotInConfig !== undefined) && (eventInConfig !== undefined)) {
+      return;
+    }
+
+    if ((eventsInConfig.indexOf(name) === -1) && (eventNotInConfig === undefined)) {
+      eventNotInConfig = eventObjects[name];
+    }
+
+    if ((eventsInConfig.indexOf(name) !== -1) && (eventInConfig === undefined)) {
+      eventInConfig = eventObjects[name];
+      findingType = events[name].type;
+      findingSeverity = events[name].severity;
+    }
+  });
+  return {
+    eventInConfig, eventNotInConfig, findingType, findingSeverity,
+  };
+}
+
+function getExpressionOperand(operator, value, expectedResult) {
+  // given a value, an operator, and a corresponding expected result, return a value that
+  // meets the expected result
+  let leftOperand;
+  /* eslint-disable no-case-declarations */
+  if (BigNumber.isBigNumber(value)) {
+    switch (operator) {
+      case '>=':
+        if (expectedResult) {
+          leftOperand = value.toString();
+        } else {
+          leftOperand = value.minus(1).toString();
+        }
+        break;
+      case '<=':
+        if (expectedResult) {
+          leftOperand = value.toString();
+        } else {
+          leftOperand = value.plus(1).toString();
+        }
+        break;
+      case '===':
+        if (expectedResult) {
+          leftOperand = value.toString();
+        } else {
+          leftOperand = value.minus(1).toString();
+        }
+        break;
+      case '>':
+      case '!==':
+        if (expectedResult) {
+          leftOperand = value.plus(1).toString();
+        } else {
+          leftOperand = value.toString();
+        }
+        break;
+      case '<':
+        if (expectedResult) {
+          leftOperand = value.minus(1).toString();
+        } else {
+          leftOperand = value.plus(1).toString();
```

Comment:
Ubernit: In all of the other cases, we either set `leftOperand` equal to `value` or we intentionally set it +-1.  In this case, we broke from that pattern.